cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1035,6 +1035,16 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
                                             configMapOperations.get(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) userOperator.getLogging()).getName()) :
                                             null) : null;
 
+                            String topicLogSettings = topicOperatorLogAndMetricsConfigMap.getData().get("log4j2.properties");
+                            String topicLoggerValue = topicLogSettings.substring(topicLogSettings.indexOf(topicOperator.TOPIC_OPERATOR_LOGGER));
+                            topicLoggerValue = topicLoggerValue.substring(topicLoggerValue.indexOf("=") + 1, topicLoggerValue.indexOf("\n"));
+                            topicOperator.setLoggingLevel(topicLoggerValue);
```

Comment:
This feels wrong. How does it work with external logging configuration? I think you should use here the whole value and hash it? Wouldn't that be easier?

Reply:
How does it work with the external? It seems to only capture the log level value and nothing else. There might be other changes when using external. Or do I read the code wrong?

Reply:
No, you are reading it well. I focused just on the log level.

Reply:
Which means that when I use external logging and change the timestamp format or the layout of the log records, it will not reload, right? Which is IMHO wrong.